lib/Borel.hs

```diff
@@ -80,8 +81,10 @@ groupMetrics :: Set Metric        -- ^ All available instances
              -> Set Metric        -- ^ Metrics to be grouped
              -> Set GroupedMetric
 groupMetrics instances metrics
-  =  let (allfs, nonfs) = _1 %~ S.toList $ S.partition (`S.member` instances) metrics
-     in  S.insert allfs $ S.map pure nonfs
+  =  let (allfs, nonfs) = (S.toList *** S.map pure) $ S.partition (`S.member` instances) metrics
+     in case allfs of
+       [] -> nonfs
+       xs -> S.insert xs nonfs
 
```

Comment:
does this really make a difference? I intended for the identity case to cause an empty list of tags, and thus achieving the same effect.